src/lightning_app/components/auto_scaler.py

```diff
@@ -0,0 +1,437 @@
+import asyncio
+import logging
+import os
+import time
+import uuid
+from itertools import cycle
+from typing import Any, Dict, List, Optional, Tuple
+
+import aiohttp
+import aiohttp.client_exceptions
+import requests
+import uvicorn
+from fastapi import FastAPI, HTTPException, Request
+from fastapi.middleware.cors import CORSMiddleware
+from fastapi.responses import RedirectResponse
+from pydantic import BaseModel
+
+from lightning_app.core.flow import LightningFlow
+from lightning_app.core.work import LightningWork
+from lightning_app.utilities.app_helpers import Logger
+from lightning_app.utilities.packaging.cloud_compute import CloudCompute
+
+MIN_REPLICA = int(os.environ.get("MUSE_MIN_WORKERS", 1))
+KEEP_ALIVE_TIMEOUT = float(os.environ.get("KEEP_ALIVE_TIMEOUT", 60))
```

Comment:
Those should be exposed as arguments to the class, not env variables. Users won't discover them.